pkg/errors/errors.go

```diff
@@ -131,6 +131,21 @@ func IsTransient(err error) bool {
 	}
 }
 
+// IsRegistryTransient returns true if err represents a transient registry error
+func IsRegistryTransient(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	switch {
+	case strings.Contains(err.Error(), "failed commit on ref") && strings.Contains(err.Error(), "500 Internal Server Error"),
```

Comment:
do you need all three messages to be in the error, or are we handling three different errors here?

Reply:
they are two different errors, one of them with two conditions